cppcache/integration/test/PdxJsonTypeTest.cpp

```diff
@@ -68,6 +68,32 @@ std::shared_ptr<Region> setupRegion(Cache& cache) {
   return region;
 }
 
+TEST(PdxJsonTypeTest, testGfshQueryJsonInstances) {
+  Cluster cluster{LocatorCount{1}, ServerCount{1}};
+  cluster.start();
+
+  auto& gfsh = cluster.getGfsh();
+  gfsh.create().region().withName("region").withType("REPLICATE").execute();
+
+  auto cache = cluster.createCache();
+  auto region = setupRegion(cache);
+  const std::string query_stmt{"SELECT * FROM /region"};
+
+  region->put("non-java-domain-class-entry",
+              cache.createPdxInstanceFactory(gemfireJsonClassName, false)
+                  .writeString("foo", "bar")
+                  .create());
+
+  EXPECT_NO_THROW(gfsh.query(query_stmt).execute());
```

Comment:
Can we really only test this via `gfsh`? What about writing a Java function to deploy and execute. That would have fewer moving parts and no output text to parse.

Reply:
Well, if possible, that would be much better, for sure. Will take a look at it :)